ksql-cli/src/main/java/io/confluent/ksql/cli/console/JLineReader.java

```diff
@@ -111,22 +120,44 @@ public String expandVar(final String word) {
       ));
     }
 
-    this.lineReader.unsetOpt(Option.HISTORY_INCREMENTAL);
-    this.history = new DefaultHistory(this.lineReader);
-
-    this.prompt = DEFAULT_PROMPT;
+    return lineReader;
   }
 
-  @Override
-  public Iterable<? extends History.Entry> getHistory() {
-    return lineReader.getHistory();
-  }
+  /**
+   * Override the default JLine 'expander' behavior so that history-referencing expressions such
+   * as '!42' or '!!' will only be processed and replaced if they occur at the beginning of an
+   * input line, even if the input line spans multiple terminal lines using the '\' separator.
```

Comment:
This comment is stale (refers to '\' separator). Also makes more sense over the expandHistory method rather than the whole class.

Reply:
`\` separator is still supported, (just discouraged), so that comment is still right. Scope of comment is... well, I've not changed it and people could argue that the reason we have `KsqlExpander` is to override the default behaviour, as the comment states.  Not my code, haven't changed it, (except to move it down), so leaving comment as-is.